src/webpack.config.js

```diff
@@ -1,38 +1,78 @@
-import webpackCleanPlugin from 'clean-webpack-plugin'
+import fs from 'fs'
+import path from 'path'
+import webpack from 'webpack'
+import CleanPlugin from 'clean-webpack-plugin'
 
-export default (context) => {
-    const config = {
-      resolve: {
-        modulesDirectories: [context, `${context}/node_modules`]
-      },
-      entry: 'tapestry-wp/dist/client.js',
-      output: {
-        path: 'public',
-        filename: 'bundle.js'
-      },
-      module: {
-        loaders: [
-          {
-            test: /\.js$/,
-            exclude: /node_modules/,
-            loader: 'babel-loader',
-            query: {
-              presets: ['es2015', 'react']
-            }
-          }
-        ]
-      },
-      plugins: [
-        new webpackCleanPlugin(['public'], { root: context, verbose: false })
-        // new webpack.optimize.UglifyJsPlugin(),
-        // new webpack.optimize.OccurrenceOrderPlugin(),
-        // new webpack.optimize.DedupePlugin(),
-        // new webpack.DefinePlugin({
-        //   'process.env': {
-        //     'NODE_ENV': JSON.stringify('production')
-        //   }
-        // })
-      ]
+const env = process.env.NODE_ENV
+
+
+export default (ctx, isTree = false) => {
+
+  const config = {
+    resolve: {
+      modules: [ctx, `${ctx}/node_modules`]
+    },
+    module: {
+      rules: [{
+        test: /\.js$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+        options: {
+          presets: ['es2015', 'react']
```

Comment:
I'm not 100% on how the 'tree-shaking' in Webpack 2 works, but if you want to take advantage, I think you may need to specify 'modules: false' for the es2015 preset? I could be wrong, but I've glanced over a few set ups and articles that say something along those lines.

Or this guy – the article is from early last year though (https://medium.com/modus-create-front-end-development/webpack-2-tree-shaking-configuration-9f1de90f3233#.3knrqfiz1) – who says `es2015-native-modules` will help enable 'tree-shaking'